source/solvers/tracer_assemblers.cc

```diff
@@ -267,25 +267,33 @@ TracerAssemblerDGCore<dim>::assemble_matrix(
       // Gather into local variables the relevant fields
       const double         diffusivity = diffusivity_vector[q];
       const Tensor<1, dim> velocity    = scratch_data.velocity_values[q];
+      const double velocity_divergence = scratch_data.velocity_divergences[q];
 
       // Store JxW in local variable for faster access;
       const double JxW = JxW_vec[q];
 
       for (unsigned int i = 0; i < n_dofs; ++i)
         {
           const auto grad_phi_T_i = scratch_data.grad_phi[q][i];
+          const auto phi_T_i      = scratch_data.phi[q][i];
 
           for (unsigned int j = 0; j < n_dofs; ++j)
             {
               const Tensor<1, dim> grad_phi_T_j = scratch_data.grad_phi[q][j];
               const auto           phi_T_j      = scratch_data.phi[q][j];
 
-              // Weak form : - D * laplacian T +  u * gradT - f=0
+              // Weak form: - D * laplacian T + u * gradT - f=0
               // Note that the advection term has been weakened for it to appear
               // explicitly in the weak form as a boundary term.
               local_matrix(i, j) += (diffusivity * grad_phi_T_i * grad_phi_T_j -
                                      grad_phi_T_i * velocity * phi_T_j) *
                                     JxW;
+
+              // This term is added to correct for the influence of the
+              // non-divergence-free velocity field on the concentration. The
+              // term is introduced in the strong form as - div u * T.
+              local_matrix(i, j) +=
+                -phi_T_i * velocity_divergence * phi_T_j * JxW;
```

Comment:
```suggestion
              // This term is added to correct for the influence of the
              // non-divergence-free velocity field on the concentration.
              local_matrix(i, j) +=
                -phi_T_i * velocity_divergence * phi_T_j * JxW;
```

It's not introduced in the strong form per say, it's because we have weakened the advection term.